hq/app/aws/ec2/EC2.scala

```diff
@@ -187,10 +192,18 @@ object EC2 {
     }
   }
 
-  private[ec2] def getVpcs(account: AwsAccount, flaggedSgs: List[SGOpenPortsDetail])(vpcsDetailsF: AmazonEC2Async => Attempt[Map[String, Vpc]])(implicit ec: ExecutionContext): Attempt[Map[String, Vpc]] = {
+  private[ec2] def getVpcs(account: AwsAccount, flaggedSgs: List[SGOpenPortsDetail], ec2Clients: Map[(String, Regions), AmazonEC2Async])(vpcsDetailsF: AmazonEC2Async => Attempt[Map[String, Vpc]])(implicit ec: ExecutionContext): Attempt[Map[String, Vpc]] = {
     Attempt.traverse(flaggedSgs.map(_.region).distinct) { region =>
-      val ec2Client = client(account, Regions.fromName(region))
-      vpcsDetailsF(ec2Client)
+      val awsRegion = Regions.fromName(region)
+      for {
+        ec2Client <- Attempt.fromOption(ec2Clients.get((account.id, awsRegion)), FailedAttempt(Failure(
```

Comment:
Why is this failure inlined here? I thought there was an `EC2.client` method that takes care of this?